### PR TITLE
virtio_serial:boot tcp_socket chardev with qmp monitor

### PR DESCRIPTION
**Pull request by nanliu-r**
ID: 1746246

Depends on [avocado-framework/#2393](https://github.com/avocado-framework/avocado-vt/pull/2393)

Signed-off-by: Nana Liu <nanliu@redhat.com>

### CODE REVIEW

**Comment by nanliu-r** [comment]
@yanan-fu  Please help to review this, thanks a lot.

**Comment by nanliu-r** [comment]
Test Pass.

Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.system_powerdown: PASS (96.66 s)
Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.system_reset_bootable: PASS (356.03 s)
Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.system_reset_during_boot: PASS (433.01 s)
Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.stop_continue.stop_cont_only: PASS (54.52 s)
Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.reboot: PASS (2863.54 s)
Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.boot: PASS (48.99 s)
Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.shutdown: PASS (51.94 s)

(10/10) repeat10.Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.boot_with_tcp_monitor: PASS (47.64 s)

File: qemu/tests/cfg/virtio_serial_basic.cfg [hunk 1]
```diff
@@ -0,0 +1,9 @@
+- boot_with_tcp_monitor: install setup image_copy unattended_install.cdrom
+    virt_test_type = qemu libvirt
```
Comment (yanan-fu) [inline qemu/tests/cfg/virtio_serial_basic.cfg]: No `libvirt`

File: qemu/tests/cfg/virtio_serial_basic.cfg [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+    type = boot
+    serial_login = no
+    restart_vm = yes
```
Comment (yanan-fu) [inline qemu/tests/cfg/virtio_serial_basic.cfg]: useless param `restart_vm`

File: qemu/tests/cfg/virtio_serial_basic.cfg [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+    kill_vm_on_error = yes
+    login_timeout = 240
+    chardev_backend_qmpmonitor1 = tcp_socket
+    chardev_backend_catch_monitor = tcp_socket
```
Comment (yanan-fu) [inline qemu/tests/cfg/virtio_serial_basic.cfg]: Do we need to test two qmp monitor ?

**Comment by yanan-fu** [comment]
Acked-by: Yanan Fu <yfu@redhat.com>

**Pull request merged**